### ISSUE TITLE
check_syntax false positive: Structure does not contain any components on CLASS-DATA

### ISSUE THREAD

**Issue by rubber-duck-53** [abaplint/abaplint#4180]
```abap
REPORT ztest_flat_assignment.

CLASS lcl_test_data DEFINITION FINAL CREATE PUBLIC.
  PUBLIC SECTION.
    TYPES: BEGIN OF ty_address,
             street  TYPE char60,
             city    TYPE char40,
             country TYPE char3,
           END OF ty_address.

    CONSTANTS:
      BEGIN OF test_constants,
        BEGIN OF data_block,
          customer_id  TYPE char10    VALUE `1000000001`,
          address_info TYPE ty_address VALUE `Unter den Linden 1                                          Berlin                                  DE `,
        END OF data_block,
      END OF test_constants.

    CLASS-DATA:
      BEGIN OF test_values,
        BEGIN OF data_block,
          customer_id  TYPE char10    VALUE `1000000001`,
          address_info TYPE ty_address VALUE `Unter den Linden 1                                          Berlin                                  DE `,
        END OF data_block,
      END OF test_values.
ENDCLASS.

```

## Expected

No findings — same behavior for `CONSTANTS` and `CLASS-DATA`

## Actual

```
ztest_flat_assignment.prog.abap[3, 1] - Structure does not contain any components (check_syntax) [E]
```
`Note:` The reported finding position points to the beginning of the class definition but is actually caused by the  `CLASS-DATA` statement.

**Comment by larshp** [comment]
@rubber-duck-53 try again, v2.120.16 published

**Comment by rubber-duck-53** [comment]
Thanks, works as expected now.
